packages/erk-shared/src/erk_shared/github/retry.py

```diff
@@ -1,71 +1,112 @@
-"""Retry utility for GitHub API calls with exponential backoff."""
+"""GitHub API retry utilities with configurable delays and exception handling."""
+
+from __future__ import annotations
 
 import sys
 from collections.abc import Callable
 from typing import TypeVar
 
 from erk_shared.gateway.time.abc import Time
 
-# Retry delays following exponential backoff pattern: [0.5, 1.0] (~1.5s max, 2 retries)
+# Default retry delays for transient error retry (exponential backoff)
 RETRY_DELAYS = [0.5, 1.0]
 
 T = TypeVar("T")
 
 
+class ShouldRetry(Exception):
```

Comment:
mkae this not an exception and instead return this directly from with_github_retry

Reply:
Resolved via /erk:pr-address at 2025-12-31 12:15 PM PST: Changed ShouldRetry from exception to frozen dataclass return value. The callback now returns ShouldRetry instead of raising it, and with_github_retry returns ShouldRetry when all attempts exhausted instead of raising it. This eliminates exception-based control flow in favor of return-value pattern.

_Addressed via `/erk:pr-address` at 2025-12-31 12:15 PST_